From: Ben Hutchings <bhutchings@solarflare.com>
Date: Thu, 27 Jun 2013 00:13:07 +0100
Subject: [PATCH v2 102/244] sfc: Remove early call to
 efx_nic_type::reconfigure_mac in efx_reset_up()
Patch-mainline: v3.12-rc1
Git-commit: ba388fdd08ff458fabc8588e565b5d031191c50a
References: bsc#909618 FATE#317521

efx_reset_up() calls efx_nic_type::reconfigure_mac once directly,
then again through efx_start_all() -> efx_start_port() ->
efx->type->reconfigure_mac().

This first call is also made too early to work properly on EF10.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/sfc/efx.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/ethernet/sfc/efx.c
+++ b/drivers/net/ethernet/sfc/efx.c
@@ -2200,8 +2200,6 @@ int efx_reset_up(struct efx_nic *efx, en
 				  "could not restore PHY settings\n");
 	}
 
-	efx->type->reconfigure_mac(efx);
-
 	efx_enable_interrupts(efx);
 	efx_restore_filters(efx);
 	efx_sriov_reset(efx);
